patches/react-native-webview+9.4.0.patch

```diff
@@ -0,0 +1,188 @@
+diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
+index 1995d87..33bd83c 100644
+--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
++++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
+@@ -2,8 +2,12 @@ package com.reactnativecommunity.webview;
+ 
+ import android.annotation.SuppressLint;
+ import android.annotation.TargetApi;
++import android.app.Activity;
++import android.app.AlertDialog;
+ import android.app.DownloadManager;
++import android.content.ActivityNotFoundException;
+ import android.content.Context;
++import android.content.DialogInterface;
+ import android.content.Intent;
+ import android.content.pm.ActivityInfo;
+ import android.content.pm.PackageManager;
+@@ -15,6 +19,7 @@ import android.os.Build;
+ import android.os.Environment;
+ import androidx.annotation.RequiresApi;
+ import androidx.core.content.ContextCompat;
++import android.text.InputType;
+ import android.text.TextUtils;
+ import android.view.Gravity;
+ import android.view.View;
+@@ -25,7 +30,9 @@ import android.webkit.ConsoleMessage;
+ import android.webkit.CookieManager;
+ import android.webkit.DownloadListener;
+ import android.webkit.GeolocationPermissions;
++import android.webkit.HttpAuthHandler;
+ import android.webkit.JavascriptInterface;
++import android.widget.LinearLayout;
+ import android.webkit.PermissionRequest;
+ import android.webkit.URLUtil;
+ import android.webkit.ValueCallback;
+@@ -35,6 +42,7 @@ import android.webkit.WebResourceResponse;
+ import android.webkit.WebSettings;
+ import android.webkit.WebView;
+ import android.webkit.WebViewClient;
++import android.widget.EditText;
+ import android.widget.FrameLayout;
+ 
+ import com.facebook.react.views.scroll.ScrollEvent;
+@@ -720,6 +728,7 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+ 
+   protected static class RNCWebViewClient extends WebViewClient {
+ 
++    protected Activity mCurrentActivity;
+     protected boolean mLastLoadFailed = false;
+     protected @Nullable
+     ReadableArray mUrlPrefixesForDefaultIntent;
+@@ -729,6 +738,10 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+       ignoreErrFailedForThisURL = url;
+     }
+ 
++    public void setCurrentActivity(Activity mCurrentActivity) {
++      this.mCurrentActivity = mCurrentActivity;
++    }
++
+     @Override
+     public void onPageFinished(WebView webView, String url) {
+       super.onPageFinished(webView, url);
+@@ -810,6 +823,49 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+         new TopLoadingErrorEvent(webView.getId(), eventData));
+     }
+ 
++    @Override
++    public void onReceivedHttpAuthRequest(WebView view,
```

Comment:
Is this Alert new functionality for Mattermost?

Reply:
If was already part of Mattermost, can you point me to where it came from?

Reply:
We were using a fork before https://github.com/mattermost/react-native-webview/ this is to help with IWA authentication